### ISSUE TITLE
Faulty tag suggestions

### ISSUE THREAD

**Issue by anand2312** [python-discord/bot#1432]
On making a typo with a tag, any additional text that was sent along with it is also sent in the "Did you mean..." embed.

![Example](https://media.discordapp.net/attachments/757877755395309609/814017613654130739/unknown.png?width=695&height=240)

The fuzzy matching also seems a bit odd here, with `pste` being matched to `site` instead of `paste`.

**Comment by Senjan21** [comment]
As to why it happens, https://github.com/python-discord/bot/blob/master/bot/exts/backend/error_handler.py#L208 shows that we simply use `.replace` on user input, we probably could simply just put the suggested command name instead of parsing whole user input and replacing it with suggestion. Would you like to work on it?

**Comment by anand2312** [comment]
I'm busy currently; anyone is free to work on this.

**Comment by dawnofmidnight** [comment]
I'm open to working on this. But to make sure I'm not misunderstanding this, you want to only use the first word and not the whole statement, correct?

**Comment by ToxicKidz** [comment]
> The fuzzy matching also seems a bit odd here, with `pste` being matched to `site` instead of `paste`.
Yeah, I guess [`difflib.get_close_matches`](https://docs.python.org/3/library/difflib.html#difflib.get_close_matches) finds multiple matches but we only chose the first one. With testing I found out that if the cutoff is `0.8`, then it only matches `paste` and not `site`.

**Comment by ToxicKidz** [comment]
It also seems that the error handler doesn't try to make suggestions for tags, and `!paste` is a tag.

**Comment by jb3** [comment]
Closed in 2025 review of open issues. We do not feel there is a large need for this feature.
